kratos/geometries/geometry.h

```diff
@@ -3461,6 +3461,18 @@ class Geometry
         return rResult;
     }
 
+    virtual int Check() const
+    {
+        KRATOS_TRY
+
+        const double domain_size = DomainSize();
+        KRATOS_ERROR_IF(domain_size <= 0.0) << "Geometry # " << std::to_string(mId) << " has non-positive size " << domain_size << std::endl;
```

Comment:
I'm very much in favor of adding the `Check` to the geometry. The point is that the error thrown with the geometry Id might be problematic as these can have the self assigned id, which would difficult debugging.

I think we should still throw the error message in both `Element` and `Condition` as these always have a meaningful id which can be backtraced in the preprocessor.

Reply:
I see your point but then we don’t need this method if we anyway throw in the element

Reply:
We can try to issue the element id in the KRATOS_CATCH of the Check of Element

@roigcarlo is this possible?
Or must the input for KRATOS_CATCH be known at compile time?